vunit/builtins.py

```diff
@@ -130,16 +135,26 @@ def _add_verification_components(self):
             raise RuntimeError("Verification component library only supports vhdl 2008 and later")
         self._add_files(VHDL_PATH / "verification_components" / "src" / "*.vhd")
 
+    def _add_library_if_not_exist(self, library_name, message):
+        """
+        Check if a library name exists in the project. If not, add it and return a handle.
+        """
+        if library_name.lower() in [
+            library.lower() for library in self._vunit_obj._project._libraries  # pylint: disable=protected-access
+        ]:
+            LOGGER.warning(message)
+            return None
+        return self._vunit_obj.add_library(library_name)
+
     def _add_osvvm(self):
         """
         Add osvvm library
         """
-        library_name = "osvvm"
-
-        try:
-            library = self._vunit_obj.library(library_name)
-        except KeyError:
-            library = self._vunit_obj.add_library(library_name)
+        library = self._add_library_if_not_exist(
+            "osvvm", "Library 'OSVVM' previously defined. Skipping addition of builtin OSVVM (2021.09)."
```

Comment:
How can we make sure that we will remember to update this message when OSVVM is bumped?

Reply:
I'm not sure... By navigating the codebase through the GitHub web, you don't get any explicit indication that OSVVM is bumped to a tag. It just points to the commit: https://github.com/OSVVM/OSVVM/tree/71faba779aebe79d3a48ce0abc0a9adc272f99ae.

We might use a test for checking out the history of OSVVM, using git for comparing the commit hash with the tags and find an exact match. Then, in the `builtins.py`, have the version moved to a variable which the check can import for comparison. However, I'm not sure that complexity is worth. Particularly, whether that should be part of any "regular" test.

We might just add a subsection to http://vunit.github.io/contributing.html about "How to bump OSVVM", explaining that we want to bump to tagged commits only, and that whoever does it should remember to update the string.

When dependabot or some other similar solution supports bumping submodules to tags only, we might actually need an automated procedure.

Reply:
I think the instruction is good enough for now. If we find out we keep missing to update the string we can come up with another solution.

Reply:
This PR now adds a subsection to the docs about bumping submodules.